schedule_lessons/dashboard/views.py

```diff
@@ -973,20 +992,47 @@ def error_check_and_save_lesson(request, lesson, context):
             context.get("no_ending_time_error") and not
             context.get("bigger_start_time_error") and not
             context.get("pending_relationship_error") and not
-            context.get("no_relationship_error") and not
-            context.get("")):
+            context.get("no_relationship_error")):
         start_time_in_local_time = datetime.datetime.combine(date, start_time,
                                                              time_difference)
         if start_time_in_local_time < datetime.datetime.now(tz=time_difference):
             context["past_lesson_error"] = (
                 "Fix starting time or date of lesson to make sure it's after "
                 "current time.")
             return context
-        context["status"] = 200
         end_time_in_local_time = datetime.datetime.combine(date, end_time,
                                                            time_difference)
         lesson.start_time = start_time_in_local_time.astimezone(UTC_ZONE)
         lesson.end_time = end_time_in_local_time.astimezone(UTC_ZONE)
+        availabilities = list(Availability.objects.filter(
+                                    profile__user=person_to_schedule_with,
+                                    day=lesson.start_time.strftime("%A")))
+        context['non_available_time_error'] = True
+        if availabilities:
+            for availability in availabilities:
+                lesson_start_t = lesson.start_time.time()
+                availability_start_t = availability.start_time.time()
+                lesson_end_t = lesson.end_time.time()
+                availability_end_t = availability.end_time.time()
+                if ((lesson_end_t < lesson_start_t and
```

Comment:
holy crap this conditional

Reply:
Yup, there's no getting around it. Working with timings is a bitch

Reply:
that sucks, but yeah looks good man